docs/architecture/adr/ADR037-per-namespace-gateways.md

```diff
@@ -0,0 +1,28 @@
+# ADR037: Per namespace istio gateways
+
+## Status
+
+Pending
+
+## Context
+
+Different tenant applications have differing ingress and egress requirements. Examples include:
+
+* mutual TLS
+* egress controls (e.g. limiting what can egress from a namespace / cluster)
+* certificate operations (renewal etc.)
+
+At present, a single ingress gateway and egress gateway are defined by the GSP. In the case of the ingress gateway, all ingress traffic is routed through this which creates a range of difficulties:
+
+* secrets relating to TLS certs (mutual or otherwise) have to be stored in `istio-system`
+* having some endpoints use mutual TLS and others using non-mutual TLS is difficult/awkward/impossible on a single gateway
+* controlling DNS entries and TLS configuration per tenant is difficult/awkward/impossible
```

Comment:
Please explain what kind of DNS entries and what kind of TLS configuration.  Possibly give examples?

Reply:
I have no idea how to address this....

Reply:
I think we're talking about any kind of DNS or TLS configuration that is different from: a service that runs on a subdomain of the cluster domain and expects to talk HTTPS which is terminated by the ALB and without client authentication"?